File: frontend/src/game/upgrades/curses/health_reduc_1.ts
import { Upgrade, type UpgradeDef } from '../Upgrade'
import { UpgradeTargetID, UpgradeStatID, RarityID, UpgradeTypeID } from '../../data/ID'

export const HealthReduc1Def: UpgradeDef = {
  id: "health_reduc_1",
  name: "Reduced Health 1",
  description: "-5 max health.",
  rarity: RarityID.Common,
  upgradeType: UpgradeTypeID.StatModifier,
  cost: 0,
  targetClass: UpgradeTargetID.Player,
  fieldInTargetClass: UpgradeStatID.MaxHealth,
  value: -5,
  isMultiplier: false,
  stackable: true,
  maxStacks: 99999,
  curse: true,
}

export class HealthReduc1 extends Upgrade {}

File: backend/app/core/upgrades/curses/health_reduc_1.py
from ..upgrade_implementation import UpgradeImplementation, UpgradeAppliers


class HealthReduc1(UpgradeImplementation):
    id = 'health_reduc_1'
    name = 'Reduced Health 1'
    description = '-5 max health.'
    rarity = 'common'
    type = 'stat_modifier'
    cost = 0

    def can_apply(self, state: dict) -> bool:
        return True

    def apply(self, appliers: UpgradeAppliers) -> None:
        value = -5
        appliers.modifiers.add_multiplier('player', 'maxHealth', value)

    def validate(self, recorded_stats: dict) -> bool:
        return True
